### PR TITLE
Remove the Draft UXarray API document

### LINKED ISSUE #427: Retire draft UXarray API document
Synchronization between [draft UXarray API](https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md) and GitHub issues requires manual handling, so it is not always done. This leaves draft UXarray API outdated and impractical to be used by the team and community.

Consensus on tasks to-do: 

- [ ] Prerequisite #316 for the next task to be handled well
- [ ] Convert draft UXArray API doc's functions to GitHub issues
        - Use #316's issue templates, GH milestones, and maybe labels for the best representation, categorization, and prioritization
- [ ] Use project board's roadmap view (or similar) to provide high level roadmap to users, similar to this draft API doc
- [ ] Retire it from GitHub repo

### PR DESCRIPTION
**Pull request by erogluorhan**
Closes #427 

## Overview
Remove the draft UXarray API (`docs/user_api/uxarray_api.md`) as discussed by the team after a number of GH issues have been created to keep track of the new methods and attributes that were referred to within that file, and address a number of relevant tasks to this removal as follows:

- Fixes the links to this file in a number of places in the repo by linking to [UXarray Milestones](https://github.com/UXARRAY/uxarray/milestones) and [UXarray roadmap](https://github.com/orgs/UXARRAY/projects/2/views/17) instead.
- Fixes usages to older `open_grid()` optional args such as `islatlon`.

## PR Checklist

**General**
- [x] An issue is linked created and linked
- [x] Add appropriate labels
- [x] Filled out the Overview section

**Documentation**
- [x] Docstrings have updated with any function changes

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -8,8 +8,10 @@ API Reference
 
 
 Below user API and internal API pages show already-implemented UXarray functionality.
-You can also check the draft `UXarray API
-<https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
+You can also check the `UXarray Milestones <https://github.com/UXARRAY/uxarray/milestones>`_
+and `UXarray roadmap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_ for a
```
Comment (philipc2) [inline docs/api.rst:12]:
```suggestion
and `UXarray Roadmap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_ for a
```

File: docs/api.rst [hunk 2]
```diff
@@ -8,8 +8,10 @@ API Reference
...
+high level understanding of UXarray's future function development milestones and roadmap.
```
Comment (philipc2) [inline docs/api.rst:13]:
```suggestion
high-level understanding of UXarray's future function development milestones and roadmap.
```

File: docs/contributing.rst [hunk 3]
```diff
@@ -65,15 +65,16 @@ Some important UXarray resources are as follows:
...
 
 * `UXarray documentation <https://uxarray.readthedocs.io/>`_
   houses significant documentation such as :ref:`quickstart`, :ref:`installation`,
   :ref:`contributing` (i.e. this document), :ref:`examples`, :ref:`tutorials`,
   and :ref:`api`.
 
-* `UXarray draft API <https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
-  tentatively shows the eventual list of UXarray functionalities. It is
+* `UXarray Milestones <https://github.com/UXARRAY/uxarray/milestones>`_ and
+  `UXarray roadmap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_ shows
```
Comment (philipc2) [inline docs/contributing.rst:76]:
```suggestion
  `UXarray Roadmap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_ shows
```

File: docs/contributing.rst [hunk 4]
```diff
@@ -188,8 +189,8 @@ select an issue to work on:
 <https://github.com/UXARRAY/uxarray/issues>`_. These issues might have been
 created from either the:
 
-* `UXarray draft API
-  <https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_ entries,
+* `UXarray Milestones <https://github.com/UXARRAY/uxarray/milestones>`_
+  and `UXarray roadmap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_,
```
Comment (philipc2) [inline docs/contributing.rst:193]:
```suggestion
  and `UXarray Roadmap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_,
```

**Review by philipc2 — CHANGES_REQUESTED** [review]
Nice work! A few small suggestions.

File: docs/internal_api/index.rst [hunk 5]
```diff
@@ -5,9 +5,10 @@ Internal API
 ############
 
 This page shows already-implemented Uxarray internal API functions. You can also
-check the draft `UXarray API
-<https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
-documentation to see the tentative whole API and let us know if you have any feedback!
+check the `UXarray Milestones <https://github.com/UXARRAY/uxarray/milestones>`_ and
+`UXarray roadmap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_ for a high
```
Comment (aaronzedwick) [inline docs/internal_api/index.rst:9]:
```suggestion
`UXarray RoadMap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_ for a high
```
Just to be consistent with Philip's suggestions

**Review by aaronzedwick — APPROVED** [review]
Looks good to me! Nice work, this is definitely going to help make the development goals clearer!

File: docs/internal_api/index.rst [hunk 6]
```diff
@@ -5,9 +5,10 @@ Internal API
 ############
 
 This page shows already-implemented Uxarray internal API functions. You can also
-check the draft `UXarray API
-<https://github.com/UXARRAY/uxarray/blob/main/docs/user_api/uxarray_api.md>`_
-documentation to see the tentative whole API and let us know if you have any feedback!
+check the `UXarray Milestones <https://github.com/UXARRAY/uxarray/milestones>`_ and
+`UXarray RoadMap <https://github.com/orgs/UXARRAY/projects/2/views/17>`_ for a high
+level understanding of UXarray's future function development milestones and roadmap.
+Please let us know if you have any feedback!
```
Comment (philipc2) [inline docs/internal_api/index.rst:11]: Could it be worth mentioning how they could provide feedback? Maybe a link to submitting an issue on our repo?

**Review by philipc2 — APPROVED** [review]
Looks good to me! One small optional suggestion, but everything else looks great!

File: docs/internal_api/index.rst (hunk 6 above)
Comment (erogluorhan) [reply docs/internal_api/index.rst:11]: Maybe not from under the API docs, but I think we should have such a guidance somewhere in our docs (I noticed there is no such information anywhere in the docs)

**Pull request merged**